### ISSUE TITLE
Connections that error in the pool cleanup queue are discarded but still count as ongoing

### ISSUE THREAD

**Issue by Healthire** [blackbeam/mysql_async#53]
If a connection is put in the connection pool cleanup queue, and that connection errors, the error is thrown away and the handle_futures method just returns Ok(()). 

src/conn/pool/mod.rs:254-266
```rust
// Handle dirty connections.
handle!(queue {
    Ok(Ready(conn)) => {
        if inner.closed {
    crate::conn::disconnect(conn);
        } else {
            returned_conns.push(conn);
        }
        handled = true;
        Ok(())
    },
    Err(_) => { Ok(()) },
});
```

Since the connection is not passed through return_conn, while the connection is removed from the queue, the ongoing count is not decremented.

The result is that:
* Connections in the queue will silently error with no way of handling or printing the error (may be intended since it's a cleanup queue anyway)
* The ongoing count will be wrong (probably not intended?)

**Comment by blackbeam** [comment]
Hi.

Thanks for report. The fact that it's a third pool related report for the last month makes me sad.
Looks like i should models this in TLA to cover all corner cases.

**Comment by blackbeam** [comment]
Fix published as [v0.19.1](https://github.com/blackbeam/mysql_async/releases/tag/v0.19.1)
